plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -487,36 +488,44 @@ int load_rules(const char* irbSet, const std::vector<std::string> &irbs, const i
-                        int i = readRuleStructAndRuleSetFromBuffer(irb.c_str(), rule_ptr);
-
-                        if (i != 0) {
-                            ruleEngineConfig.ruleEngineStatus = INITIALIZED;
-                            return i;
+                try {
+                    in_memory_rulebases stored_rulebases(irbs);
+                    for (auto const& irb : irbs) {
+                        try {
+                            auto* rule_ptr = const_cast<char*>(&stored_rulebases.get_rulebase(irb)[0]);
+                            int i = readRuleStructAndRuleSetFromBuffer(irb.c_str(), rule_ptr);
+
+                            if (i != 0) {
+                                ruleEngineConfig.ruleEngineStatus = INITIALIZED;
+                                return i;
+                            }
+                        }
+                        catch (std::out_of_range& ex) {
+                            logger::rule_engine::warn("load_rules: attempted to access nonexistent rulebase [{}]", irb);
                         }
                     }
-                    catch (std::out_of_range& ex) {
-                        logger::rule_engine::warn("load_rules: attempted to access nonexistent rulebase [{}]", irb);
-                    }
-                }
 
-                createCoreRuleIndex( );
+                    createCoreRuleIndex();
 
-                /* set max timestamp */
-                time_type_set( ruleEngineConfig.timestamp, timestamp );
-                std::string hash;
-                int ret = hash_rules_with_copy(irbs, hash, stored_rulebases);
-                if (ret >= 0) {
-                    rstrcpy(ruleEngineConfig.hash, hash.c_str(), CHKSUM_LEN);
-                } else {
+                    /* set max timestamp */
+                    time_type_set(ruleEngineConfig.timestamp, timestamp);
+                    std::string hash;
+                    int ret = hash_rules_with_copy(irbs, hash, stored_rulebases);
+                    if (ret >= 0) {
+                        rstrcpy(ruleEngineConfig.hash, hash.c_str(), CHKSUM_LEN);
+                    }
+                    else {
+                        ruleEngineConfig.ruleEngineStatus = INITIALIZED;
+                        return ret;
+                    }
+                    snprintf(ruleEngineConfig.ruleBase, sizeof(ruleEngineConfig.ruleBase), "%s", irbSet);
                     ruleEngineConfig.ruleEngineStatus = INITIALIZED;
-                    return ret;
                 }
-                snprintf( ruleEngineConfig.ruleBase, sizeof( ruleEngineConfig.ruleBase ), "%s", irbSet );
-                ruleEngineConfig.ruleEngineStatus = INITIALIZED;
+                catch (const irods::exception& e) {
+                    logger::rule_engine::error(
+                        "{} encountered an exception in file {}:{}", __PRETTY_FUNCTION__, __FILE__, e.what());
```

Comment:
A couple of things to consider here...

1. `irods::exception::what()` prints out a whole stacktrace and some other pieces that do not format particularly well in our new JSON-based logs. If you think it is helpful to include the stacktrace here for purposes of debugging (especially for user deployments) in the future, then this is fine. If you just want to print the message from the exception and don't care about the other bits, we can use `irods::exception::client_display_what()`.
2. Could we rephrase this log message slightly so that it's clear(er) that the "file" being referred to here is the source code? Maybe it's just me, but I thought this could have perhaps been referring to a rulebase file. In context, it clearly is not, but I figured I would mention it anyway. Also, if we end up using `what()` instead of `client_display_what()`, I believe it already prints out the file and line number in that extra information.

Reply:
>     2. Could we rephrase this log message slightly so that it's clear(er) that the "file" being referred to here is the source code?

Do you have any wording suggestions? Maybe something like "...in source file..." or "...in server source file..."?